### ISSUE TITLE
Error when setting a User struct in the error context

### ISSUE THREAD

**Issue by jaimeiniesta** [elixir-error-tracker/error-tracker#65]
Hi, I'm experimenting with setting information about the current user in the ErrorTracker context. I use Pow for authentication so I have a plug that does this:

```elixir
defmodule MyAppWeb.Plugs.ErrorTrackerContext do
  def init(opts), do: opts

  def call(conn, _opts) do
    ErrorTracker.set_context(%{current_user: conn.assigns.current_user})

    conn
  end
end
```

where that `current_user` is a `User` struct.

If an error is recorded, it appears on the error index but when I show it, it fails with:

```
[error] GenServer #PID<0.1459.0> terminating
** (MatchError) no match of right hand side value: []
    (error_tracker 0.2.2) lib/error_tracker/web/live/show.ex:35: ErrorTracker.Web.Live.Show.handle_params/3
    (phoenix_live_view 0.20.14) lib/phoenix_live_view/utils.ex:462: anonymous fn/5 in Phoenix.LiveView.Utils.call_handle_params!/5
    (telemetry 1.2.1) /Users/jaime/code/rocketvalidator/rocket/deps/telemetry/src/telemetry.erl:321: :telemetry.span/3
    (phoenix_live_view 0.20.14) lib/phoenix_live_view/channel.ex:567: Phoenix.LiveView.Channel.maybe_call_mount_handle_params/4
    (phoenix_live_view 0.20.14) lib/phoenix_live_view/channel.ex:1178: Phoenix.LiveView.Channel.verified_mount/8
    (phoenix_live_view 0.20.14) lib/phoenix_live_view/channel.ex:84: Phoenix.LiveView.Channel.handle_info/2
    (stdlib 4.3.1.3) gen_server.erl:1123: :gen_server.try_dispatch/4
    (stdlib 4.3.1.3) gen_server.erl:1200: :gen_server.handle_msg/6
    (stdlib 4.3.1.3) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

I'm aware that storing the whole `User` is overkill so I changed it to only store the ID. A map with some keys also works fine, it only fails in the above case, which I don't need but I thought you'd like to know.

**Comment by jaimeiniesta** [comment]
Thanks for the fix in 0.2.3 - I've checked that this solves the issue.

When trying to set an invalid context, a warning would be better, or just ignoring the context but allowing the error to be saved, without the invalid context. Otherwise, it might go unnoticed, if you set an invalid context no errors will be saved. Just an idea, if that's not possible then documentation is good enough.

**Comment by odarriba** [comment]
I like the idea. I'm going to take a look at how complex would it be to detect those cases and store the occurrence without the context while emitting a warning message.

**Comment by odarriba** [comment]
The improved approach to this has been released in v0.2.4

More details on how it behaves now in https://github.com/elixir-error-tracker/error-tracker/pull/71
